pkg/kubectl/cmd/clusterinfo/clusterinfo_dump_test.go

```diff
@@ -68,3 +79,109 @@ func TestSetupOutputWriterFile(t *testing.T) {
+	}
+
+	encodeRespForAppsGV := func(obj runtime.Object) io.ReadCloser {
+		return encodeResp(obj, appsv1.SchemeGroupVersion)
+	}
+
+	tests := map[string]struct {
+		expNsListReq []string
+
+		dumpAllNs        bool
+		populateCmdFlags func(f *flag.FlagSet)
+	}{
+		"should dump default namespaces": {
+			expNsListReq: []string{"kube-system", "default"},
+
+			populateCmdFlags: func(f *flag.FlagSet) {
+				// use default options
+			},
+		},
+
+		"should dump requested namespaces": {
+			expNsListReq: []string{"qa", "production"},
+
+			populateCmdFlags: func(f *flag.FlagSet) {
+				f.Set("namespaces", "qa,production")
+			},
+		},
+
+		"should dump all available namespaces": {
+			expNsListReq: []string{"qa", "production", "test", "kube-system"},
+
+			dumpAllNs: true,
+			populateCmdFlags: func(f *flag.FlagSet) {
+				f.Set("all-namespaces", "true")
+			},
+		},
+	}
+
+	for tn, tc := range tests {
+		t.Run(tn, func(t *testing.T) {
+			expListReq := map[string]io.ReadCloser{
+				"/api/v1/nodes": encodeRespForCoreGV(&corev1.NodeList{}),
+			}
+
+			for _, ns := range tc.expNsListReq {
+				expListReq[fmt.Sprintf("/api/v1/namespaces/%s/events", ns)] = encodeRespForCoreGV(&corev1.EventList{})
+				expListReq[fmt.Sprintf("/api/v1/namespaces/%s/replicationcontrollers", ns)] = encodeRespForCoreGV(&corev1.ReplicationControllerList{})
+				expListReq[fmt.Sprintf("/api/v1/namespaces/%s/services", ns)] = encodeRespForCoreGV(&corev1.ServiceList{})
+				expListReq[fmt.Sprintf("/api/v1/namespaces/%s/pods", ns)] = encodeRespForCoreGV(&corev1.PodList{})
+
+				expListReq[fmt.Sprintf("/apis/apps/v1/namespaces/%s/daemonsets", ns)] = encodeRespForAppsGV(&appsv1.DaemonSetList{})
+				expListReq[fmt.Sprintf("/apis/apps/v1/namespaces/%s/deployments", ns)] = encodeRespForAppsGV(&appsv1.DeploymentList{})
+				expListReq[fmt.Sprintf("/apis/apps/v1/namespaces/%s/replicasets", ns)] = encodeRespForAppsGV(&appsv1.ReplicaSetList{})
+			}
+
+			if tc.dumpAllNs { // register expected namespaces
+				var items []corev1.Namespace
+				for _, nsName := range tc.expNsListReq {
+					items = append(items, corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: nsName}})
+				}
+				expListReq["/api/v1/namespaces"] = encodeRespForCoreGV(&corev1.NamespaceList{
+					Items: items,
+				})
+			}
+
+			tf.Client = &fake.RESTClient{
+				NegotiatedSerializer: ns,
+				Client: fake.CreateHTTPClient(func(req *http.Request) (*http.Response, error) {
+					respBody, found := expListReq[req.URL.Path]
+					if !found {
+						t.Fatalf("unexpected request: %#v\n%#v", req.URL, req)
+					}
+
+					return &http.Response{
+						StatusCode: http.StatusOK,
+						Header:     cmdtesting.DefaultHeader(),
+						Body:       respBody,
+					}, nil
+				}),
+			}
+			tf.ClientConfigVal = cmdtesting.DefaultClientConfig()
+
+			ioStreams, _, _, _ := genericclioptions.NewTestIOStreams()
+			cmd := NewCmdClusterInfoDump(tf, ioStreams)
+			tc.populateCmdFlags(cmd.Flags())
+
+			cmd.Run(cmd, []string{})
```

Comment:
You're over-complicating the test, you have `ClusterInfoDumpOptions` struct available, use that to inject whatever you need for the test (iow. namespaces).

Reply:
The structs are here specifically to make the testing easier, not harder. The fact that some commands are using `cmd.Flags.Set` is old artifact. The new required pattern is to use that struct.